libs/libarchfpga/src/arch_util.cpp

```diff
@@ -194,10 +194,16 @@ void free_arch(t_arch* arch) {
         vtr::free(arch->model_library[2].outputs->name);
         delete[] arch->model_library[2].outputs;
         vtr::free(arch->model_library[3].name);
-        vtr::free(arch->model_library[3].inputs->name);
+        vtr::free(arch->model_library[3].inputs[0].name);
```

Comment:
Probably should also replace the vtr:free with delete calls, and wherever these are malloc'ed should instead new them (we've moved almost entirely to new/delete, but this file is a mix of free and delete[]).  If you want to do that clean up go ahead, but if not it can be left for a later time / someone else.

Reply:
This cannot be easily shortened to single for loop (as long as we're talking about whole library) because each model has different configuration of inputs and outputs.

I also did change all remaining `free`/`calloc`/`malloc` calls to `delete`/`new` :)